### PR TITLE
Automate the setup of a new application

### PR DESCRIPTION
**Pull request by bokay**
Rather than doing setup tasks manually, the setup process would copy the application and rename, and remove files that are not applicable to the new application.

### Notice:
`./bin/project.init -a music-accent -g civilcode/music-accent -r .`

#### Parameters:
-a application_name (can contain dashes)
-g github repo (eg: `civilcode/magasin_platform`)
-r magasin's root_path (`.` if run from the project root)
-Y confirm / force yes
-Q generates the structure only

### CODE REVIEW

**Comment by bokay** [comment]
TODO:
- [ ] Exclude copy of migration files
- [ ] Exclude copy of domain modules (?)

File: bin/project.init [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+## Parse arguments passed to the command line
+
+while getopts ":a:r:g:YQ" option; do
+    case "$option" in
+    a)
+        APP_NAME=$OPTARG
+        APP_NAME_CAMEL=$(dash_case_to_camel_case $APP_NAME)
+        APP_NAME_UPPERCASE=$(dash_case_to_upper_case $APP_NAME)
+        APP_NAME_UNDERSCORE=$(dash_case_to_underscore_case $APP_NAME)
```
Comment (nicholasjhenry) [inline bin/project.init]: So we need the client and app name to generate the structure. This is the tree we should have, based on the client `acme` and the app `cool_store`:

```
acme_platform
  apps
    cool_store
    cool_store_web
    acme_ex
```

Comment (bokay) [reply bin/project.init]: In order to do that, we'd need to have that distinction in the reference app as well.
Right now, everything is named `magasin`.
That's possibly a big change here that would deserve its own PR. What do you think?

Comment (nicholasjhenry) [reply bin/project.init]: Yeah, good call -- and that would make the convention more obvious in the reference app. 👍

File: bin/project.init [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    if [ "$OS_PLATFORM" = "Darwin" ]; then
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i '' -e "s/Magasin/$APP_NAME_CAMEL/g" $file_path
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i '' -e "s/magasin/$APP_NAME_UNDERSCORE/g" $file_path
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i '' -e "s/MAGASIN/$APP_NAME_UPPERCASE/g" $file_path
+    else
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i -e "s/Magasin/$APP_NAME_CAMEL/g" $file_path
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i -e "s/magasin/$APP_NAME_UNDERSCORE/g" $file_path
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i -e "s/MAGASIN/$APP_NAME_UPPERCASE/g" $file_path
+    fi
+done
```
[the remark this answers is not in the document]
Comment (bokay) [reply bin/project.init]: This should be fixed with a fix in the OS detection.

File: bin/project.init [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+git init .
+git add .
+git commit -m "🌱"
+git remote add origin https://github.com/$GITHUB_REPO.git
+
+#Return to the original project
+cd -
+
+echo "SUCCESS !"
+echo "Your new project has been generated here: $NEW_PROJECT_ROOT"
```
[the remark this answers is not in the document]
Comment (bokay) [reply bin/project.init]: We can run `dea mix phx.server` right away. I've added the `mix project.setup` in the `bin/setup.docker` script (it was commented)

File: bin/project.init [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i '' -e "s/Magasin/$APP_NAME_CAMEL/g" $file_path
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i '' -e "s/magasin/$APP_NAME_UNDERSCORE/g" $file_path
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i '' -e "s/MAGASIN/$APP_NAME_UPPERCASE/g" $file_path
+    else
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i -e "s/Magasin/$APP_NAME_CAMEL/g" $file_path
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i -e "s/magasin/$APP_NAME_UNDERSCORE/g" $file_path
+        LANG=C LC_CTYPE=C LC_ALL=C sed -i -e "s/MAGASIN/$APP_NAME_UPPERCASE/g" $file_path
+    fi
+done
+
```
[the remark this answers is not in the document]
Comment (bokay) [reply bin/project.init]: There were some typos in the list of exclusion. This should be fixed as well.

**Pull request merged**